src/radix_connect/well_known_client/client.rs

```diff
@@ -0,0 +1,141 @@
+use crate::prelude::*;
+
+const SUFFIX_WELL_KNOWN_FILE: &str = ".well-known/dapp.json";
+
+pub struct WellKnownClient {
+    pub http_client: HttpClient,
+}
+
+impl WellKnownClient {
+    pub fn new(http_client: HttpClient) -> Self {
+        Self { http_client }
+    }
+
+    pub fn new_with_network_antenna(
+        network_antenna: Arc<dyn NetworkAntenna>,
+    ) -> Self {
+        Self::new(HttpClient::new(network_antenna))
+    }
+}
+
+impl NetworkRequest {
+    fn get_well_known_request(url: Url) -> Self {
```

Comment:
lacking documentation. the `get...request` makes it sound like this is a getter, but this is a constructor! I would drop the `_request, it is being called on `NetworkRequest, so seems superflous.